pkgs/tools/nix/nixos-render-docs/src/tests/test_html.py

```diff
@@ -167,7 +167,7 @@ def test_full() -> None:
           <span class="citerefentry"><span class="refentrytitle">man</span>(1)</span>
          </a> reference
         </p>
-        <p><a id="b" />some <a id="a" />nested anchors</p>
+        <p><a id="b"></a>some <a id="a"></a>nested anchors</p>
```

Comment:
I don't understand this test. Isn't this a nested anchor?

```html
<a>
   Anchor
   <a>
     Nested Anchor
   </a>
</a>
```

We only have consecutive empty anchors in front of regular paragraphs:

```html
<p>
<a></a>
Some text
<a></a>
Some more text
</p>
```

Reply:
It is kinda weird because of some confusing design choices made by nixos-render-docs:

It supports the syntax `[]{#some-id-here}`, which can have text inside it (but doesn't make sense to, as you'll see shortly): `[some text]{#some-id-here}`.

This is different from the usual "link syntax": `[some text](#some-id-here)`.

The first syntax (the one with `{}`) **declares** an element with an `id`. This is used so we can jump to a specific point in the document by going to `https://nixos.org/manual/.../stable#id`.

The "link syntax" **adds** a link to the text inside: `[some text](#some-id-here)` becomes `<a href="#some-id-here>some text</a>`.

The confusing choice is that the other syntax *also* creates an `<a>` tag: `<a id="some-id-here"></a>`, but in this case, the `<a`> tag isn't used as a link  to anything. It is merely used as an element with an `id`. Note that the `<a>` doesn't have an `href` attribute. It is essentially useless. It could've been a `<span>` or some other element.

Because of that, it doesn't make sense to have text inside the `[]{#some-id-here}`, because we're not adding any links there. We're just marking that place in the text because we want to be able to jump to it by going to `https://nixos.org/...#some-id-here`.

Reply:
cc @pennae to sanity check this and the fixes I made in nrd

Reply:
Fixed to emit `span` tags instead! Thanks!